### PR TITLE
[DOCS] Add note about patterns and stored paths, #4160

### PR DESCRIPTION
**Pull request by fantasya-pbem**
Add a section that explains storage of relative paths and how patterns can be used arbitrarily since Borg 1.2.

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -2164,7 +2176,7 @@ def do_break_lock(self, args, repository):
             $ borg create -e '/home/*/junk' backup /
 
             # Exclude the contents of '/home/user/cache' but not the directory itself:
-            $ borg create -e /home/user/cache/ backup /
+            $ borg create -e home/user/cache/ backup /
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2179]: hmm, this (`/home/...`) never matched before your changes, right?

File: src/borg/archiver.py [hunk 2]
```diff
@@ -2179,8 +2191,8 @@ def do_break_lock(self, args, repository):
             /home/*/junk
             *.tmp
             fm:aa:something/*
-            re:^/home/[^/]\\.tmp/
-            sh:/home/*/.thumbnails
+            re:^home/[^/]\\.tmp/
+            sh:home/*/.thumbnails
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2195]: same here!?

File: src/borg/archiver.py (hunk 1 above)
Comment (fantasya-pbem) [reply src/borg/archiver.py:2179]: Before, in ``create`` it matched only as `/home`.

Comment (fantasya-pbem) [reply src/borg/archiver.py:2179]: And: Before, in ``extract`` it matched only as `home`.

File: src/borg/archiver.py (hunk 2 above)
Comment (fantasya-pbem) [reply src/borg/archiver.py:2195]: I removed the / to show that it can be used or be left out.

File: src/borg/archiver.py (hunk 1 above)
Comment (ThomasWaldmann) [reply src/borg/archiver.py:2179]: uh, right.

File: src/borg/archiver.py (hunk 2 above)
Comment (ThomasWaldmann) [reply src/borg/archiver.py:2195]: ok

**Pull request merged**